internal/scenes/game/systems/menu.go

```diff
@@ -0,0 +1,173 @@
+package systems
+
+import (
+	"github.com/ebitenui/ebitenui"
+	"github.com/ebitenui/ebitenui/image"
+	"github.com/ebitenui/ebitenui/widget"
+	"github.com/golang/freetype/truetype"
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/inpututil"
+	"github.com/ubootgame/ubootgame/internal/scenes/game/components"
+	"github.com/yohamta/donburi"
+	"github.com/yohamta/donburi/ecs"
+	"golang.org/x/image/font"
+	"golang.org/x/image/font/gofont/goregular"
+	"image/color"
+	"log"
+	"os"
+)
+
+type menusystem struct {
+	UI        *ebitenui.UI
+	menuEntry *donburi.Entry
+}
+
+var Menu = &menusystem{
+	UI: &ebitenui.UI{Container: setupMenu()},
+}
+
+func setupMenu() *widget.Container {
+
+	// construct a new container that serves as the root of the UI hierarchy
+	rootContainer := widget.NewContainer(
+		// the container will use a plain color as its background
+		widget.ContainerOpts.BackgroundImage(image.NewNineSliceColor(color.NRGBA{0, 0, 0, 175})),
+
+		// the container will use an anchor layout to layout its single child widget
+		widget.ContainerOpts.Layout(widget.NewAnchorLayout()),
+	)
+
+	menuContainer := widget.NewContainer(
+		widget.ContainerOpts.WidgetOpts(
+			widget.WidgetOpts.LayoutData(widget.AnchorLayoutData{
+				HorizontalPosition: widget.AnchorLayoutPositionCenter,
+				VerticalPosition:   widget.AnchorLayoutPositionCenter,
+			}),
+		),
+		widget.ContainerOpts.Layout(
+			widget.NewRowLayout(
+				widget.RowLayoutOpts.Direction(widget.DirectionVertical),
+				widget.RowLayoutOpts.Spacing(30),
+			),
+		),
+	)
+
+	newGameButton := createMenuButton("New Game", func(args *widget.ButtonClickedEventArgs) {
+		println("options button clicked")
+	})
+	optionsButton := createMenuButton("Options", func(args *widget.ButtonClickedEventArgs) {
+		println("options button clicked")
+	})
+	quitButton := createMenuButton("Quit", func(args *widget.ButtonClickedEventArgs) {
+		os.Exit(1)
+	})
+
+	// add the newGameButton as a child of the container
+	menuContainer.AddChild(newGameButton)
+	menuContainer.AddChild(optionsButton)
+	menuContainer.AddChild(quitButton)
+	rootContainer.AddChild(menuContainer)
+
+	return rootContainer
+}
+
+func createMenuButton(buttonText string, handler func(args *widget.ButtonClickedEventArgs)) *widget.Button {
+	buttonImage, _ := loadButtonImage()
```

Comment:
maybe we shouldn't be loading the font for each new button?

Reply:
no, we can do it once, in an `init()` function, just like in `utility/debug.go`

Reply:
maybe fonts should even be a resource, managed by the resource registry in `utility/resources/registry.go`

Reply:
Images can (should?) also be managed by the resource registry, I think